apollo-router-core/src/json/map.rs

```diff
@@ -0,0 +1,839 @@
+//! wrapper for `IndexMap`
+//!
+//! forked from `serde_json::map::Map`
+use super::value::Value;
+use indexmap::{self, IndexMap};
+use serde::de;
+use std::{
+    borrow::Borrow,
+    fmt::{self, Debug},
+    hash::Hash,
+    iter::FusedIterator,
+    mem, ops,
+};
+
+/// Represents a JSON key/value type.
+pub struct Map<K, V> {
+    map: MapImpl<K, V>,
+}
```

Comment:
:yarn: I doubt you need this entire file at all. serde_json::Map already has generics. You can probably do:

```rust
type Object = serde_json::Map<String, super::value::Value>;
```

And it will work just as you expect.

Reply:
it is needed because the type itself is generic, but everything in the [map module](https://github.com/serde-rs/json/blob/master/src/map.rs) is implemented on `Map<String, serde_json::Value>`, they're not generic.

Here I replace the `Value` type, but I am also planning to replace the `String` key with a `ByteString`

Reply:
Right I missed that!